easybuild/easyblocks/i/icc.py

```diff
@@ -105,53 +112,81 @@ def sanity_check_step(self):
-                'LD_LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'MANPATH': ['man', 'share/man', 'man/en_US'],
-                'IDB_HOME': ['bin/intel64']
-            }
+            if LooseVersion(self.version) < LooseVersion("2016"):
+                # 64-bit toolkit
+                libpaths = ['compiler/lib/intel64', 'lib/intel64', 'debugger/ipt/intel64/lib', 'ipp/lib/intel64', 'tbb/lib/intel64']
+                dirmap = {
+                    'PATH': ['bin/intel64', 'tbb/bin/intel64', 'ipp/bin/intel64', 'debugger/gdb/intel64/bin'],
+                    'LD_LIBRARY_PATH': libpaths, 
+                    'LIBRARY_PATH': libpaths,
+                    'MANPATH': ['man', 'share/man', 'man/en_US', 'debugger/gdb/intel64/share/man'],
+                    'CPATH': ['ipp/include', 'tbb/include'],
+                    'INTEL_PYTHONHOME': ['debugger/python/intel64']
+                }
+            else:
+                # 64-bit toolkit
+                libpaths = ['daal/../compiler/lib/intel64_lin', 'daal/../tbb/lib/intel64_lin/gcc4.4', 'daal/lib/intel64_lin', '%s/libipt/intel64/lib' % debuggerpath,'tbb/lib/intel64/gcc4.4', 'mkl/lib/intel64', 'ipp/lib/intel64', 'ipp/../compiler/lib/intel64', 'compiler/lib/intel64']
+                dirmap = {
+                    'PATH': ['mpi/intel64/bin', 'ipp/bin/intel64', '%s/gdb/intel64/bin' % debuggerpath, 'bin/intel64'],
+                    'LD_LIBRARY_PATH': libpaths,
+                    'LIBRARY_PATH': libpaths,
+                    'MANPATH': ['man/common', 'man/en_US', 'debugger/gdb/intel64/share/man'],
+                    'CPATH': ['daal/include', 'tbb/include', 'mkl/include', 'ipp/include'],
+                    'INTEL_PYTHONHOME': ['%s/python/intel64' % debuggerpath],
+                    'DAALROOT': ['daal'],
+                    'TBBROOT': ['tbb'],
+                    'IPPROOT': ['ipp'],
+                    'CLASSPATH': ['daal/lib/daal.jar'],
+                }
+
 
         # in recent Intel compiler distributions, the actual binaries are
         # in deeper directories, and symlinked in top-level directories
         # however, not all binaries are symlinked (e.g. mcpcom is not)
-        if os.path.isdir("%s/composerxe-%s" % (self.installdir, self.version)):
-            prefix = "composerxe-%s" % self.version
+        # more recent versions of the Intel Compiler (2013.sp1 and newer)
+        if os.path.isdir("%s/composer_xe_%s" % (self.installdir, self.version)):
+            prefix = "composer_xe_%s" % self.version
             oldmap = dirmap
             dirmap = {}
             for k, vs in oldmap.items():
                 dirmap[k] = []
-                if k == "LD_LIBRARY_PATH":
-                    prefix = "composerxe-%s/compiler" % self.version
-                else:
-                    prefix = "composerxe-%s" % self.version
+                prefix = "composer_xe_%s" % self.version
```

Comment:
OK, then make sure the `if`/`else` version is retained, there's probably a good reason it's there.